ion/integration_test.go

```diff
@@ -0,0 +1,807 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package ion
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+const goodPath = "../ion-tests/iontestdata/good"
+const badPath = "../ion-tests/iontestdata/bad"
+const equivsPath = "../ion-tests/iontestdata/good/equivs"
+const nonEquivsPath = "../ion-tests/iontestdata/good/non-equivs"
+
+type testingFunc func(t *testing.T, path string)
+
+type item struct {
```

Comment:
If we don't already have one, could you open an issue to track replacing this type with something more formal when we get around to it?

Reply:
[reader struct in Reader.go](https://github.com/amzn/ion-go/blob/master/ion/reader.go#L188) is a supreset of this struct but I did not want to use that.

I do not understand why we need a more formal type for this. ```ionItem``` is being used only for this test, should we create a public type?

Reply:
> I do not understand why we need a more formal type for this. ionItem is being used only for this test, should we create a public type?

Sorry, to clarify: I don't think we need a more formal type for this--it's fine as-is. However, I expect that at some point in the (distant?) future, `ion-go` will have a public-facing formal type that represents any Ion value. (Similar to `ion-java`'s `IonValue` class.) If/when we add that, we should replace this so we're not maintaining two data types for the same thing.

Reply:
Oh I see. Thanks for the explanation. We also had this conversation internally. Glad you brought that up too.
Issue #46